adlfs/spec.py

```diff
@@ -1637,6 +1658,10 @@ def download(self, rpath, lpath, recursive=False, **kwargs):
         """Alias of :ref:`FilesystemSpec.get`."""
         return self.get(rpath, lpath, recursive=recursive, **kwargs)
 
+    def sign(self, path, expiration=100, **kwargs):
+        """Create a signed URL representing the given path."""
+        return self.url(path, expires=expiration, **kwargs)
```

Comment:
s3fs uses `url()` for generating signed URLs, which I'm assuming is why this was originally implemented as `url()` in adlfs, but the top level fsspec defines actually defines `sign()` for this purpose:
https://github.com/fsspec/filesystem_spec/blob/ac589d8318d761ab6e198e00ba6cbd6c663732d7/fsspec/spec.py#L1536C5-L1537C59

(`s3fs.sign()` just calls `s3fs.url()`)